src/lib/Components/Inbox/Conversations/Preview/InvoicePreview.tsx

```diff
@@ -99,29 +111,77 @@ const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ state }) => {
   }
 }
 
-export const InvoicePreview: React.SFC<Props> = ({ invoice, onSelected }) =>
-  <TouchableHighlight onPress={onSelected} underlayColor={colors["gray-light"]}>
-    <Container>
-      <Icon source={require("../../../../../../images/payment_request.png")} />
-      <TextContainer>
-        <PaymentRequest>Payment request</PaymentRequest>
-        <CostLabel>
-          {invoice.total}
-        </CostLabel>
-      </TextContainer>
-      <TextContainer>
-        <InvoiceStateButton state={invoice.state} />
-      </TextContainer>
-    </Container>
-  </TouchableHighlight>
-
-export default createFragmentContainer(
+interface State {
```

Comment:
Basically, the component is now a proper class, and there's some state.

We hold `subscription` (so we can remove the listener), and also `hasReceivedNotification`. This is so that we can optimistically re-render in the 'Paid' state (even though we re-fetch).

Note: once this state is set (which we are assuming based on the notification), why do we need to re-fetch? If we do want to update the UI right away, but then also re-fetch, I'll need to tweak something. Currently `hasReceivedNotification` state being set will always render 'paid'.

Reply:
> Note: once this state is set (which we are assuming based on the notification), why do we need to re-fetch? 

It’s a good question. Payments are such a sensitive topic that I’d prefer we render the server reflection of it where possible, just in case something goes wrong somewhere after all and we don’t give the wrong impression. What do you think?

> Currently `hasReceivedNotification` state being set will always render 'paid'.

You can reset `hasReceivedNotification` in `componentWillReceiveProps`.